R/verb-joins.R

```diff
@@ -639,25 +711,15 @@ join_two_table_alias <- function(names, from) {
   out
 }
 
+# TODO this can probably be removed
 check_suffix <- function(x, call) {
   check_character(x, arg = "suffix", call = call)
   vctrs::vec_assert(x, character(), size = 2, arg = "suffix", call = call)
   list(x = x[1], y = x[2])
 }
 
-add_suffixes <- function(x, y, suffix) {
-  if (identical(suffix, "")) {
-    return(x)
+check_join_by_supported <- function(by, call = caller_env()) {
+  if (any(by$filter != "none")) {
+    cli_abort("Rolling joins aren't supported on database backends.", call = call)
```

Comment:
It seems likely we'll need to relax this on a per backend basis at some point, but I don't think we need to do it prospectively.

Reply:
I don't think there are many backends that support ASOF style joins though. None of the common ones like Postgres, SQLite, MS SQL Server, etc

Reply:
But at least there are usually some workarounds to get the same results. In case I need a rolling join in the database I might just implement the workaround in dbplyr 😄